File: C2/PDOO/Practicas/DeepSpaceJava/src/deepspace/ShieldBooster.java
package deepspace;



/**
 *
 * @author Pedro Pablo Ruiz Huertas y Juan Antonio Villegas Recio
 */
public class ShieldBooster implements Copyable <ShieldBooster>, CombatElement {
    private String name;
    private float boost;
    private int uses;
    
    public ShieldBooster(String new_name, float new_boost, int new_uses){
        name=new_name;
        boost=new_boost;
        uses=new_uses;
    }
    
    public ShieldBooster(ShieldBooster s){
        this.name=s.name;
        this.boost=s.boost;
        this.uses=s.uses;
    }
    
    public float getBoost(){
        return boost;
    }
    
    @Override
    public int getUses(){
        return uses;
    }
    
    ShieldToUI getUIversion(){
        return new ShieldToUI(this);
    }
    
    @Override
    public float useIt(){
        if(uses>0){
            uses--;
            return boost;
        }
        return (float) 1.0;
    }
    
    public ShieldBooster copy(){
        return new ShieldBooster(this);
    }
    
    @Override
    public String toString(){
        return "Nombre: "+this.name+" Potencia: "+this.getBoost()+" Usos: "+this.getUses()+"  ";
    }
}

File: C2/PDOO/Practicas/DeepSpaceRuby/lib/ShieldBooster.rb

#@author Pedro Pablo Ruiz Huertas y Juan Antonio Villegas Recio

#encoding:utf-8

require_relative "ShieldToUI"

module Deepspace
  class ShieldBooster
   attr_reader :boost, :uses, :name
   
   public
   
   def initialize(n, b, u)
      @name = n
      @boost = b
      @uses = u
   end
    
   def self.newCopy(s)
     new(s.name, s.boost, s.uses)
   end
   
   def getUIversion
     return ShieldToUI.new(self)
   end
   
   def useIt()
     if @uses > 0
       @uses = @uses - 1
       return @boost
     else
       return 1.0
     end
   end
   
   def to_s
     return "Nombre: #{@name}  Aumento: #{@boost}  Usos: #{@uses}\n"
   end
   
   
  end
end
